### ISSUE TITLE
Parser Bug While Entering Code

### ISSUE THREAD

**Issue by VRNocturne** [godotengine/godot#80266]
### Godot version

v4.1.stable.official [970459615]

### System information

Godot v4.1.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3070 Ti (NVIDIA; 31.0.15.3179) - AMD Ryzen 9 5900X 12-Core Processor (24 Threads)

### Issue description

While typing code (a match statement in case it's relevant) into the IDE, I started seeing parser errors.

### Steps to reproduce

Opened my project

Started editing code

ran project once 

Continued editing code, this is when I started seeing the parser errors

In the test project (attached), it just started happening while making the const values 

### Minimal reproduction project

[GodotTest.zip](https://github.com/godotengine/godot/files/12262760/GodotTest.zip)

**Comment by rsubtil** [comment]
Can you send some screenshots and/or a video clip of the issue you're facing? I tried messing around on the linked project, but I can't seem to trigger any parser errors.

Also I've tested with v4.1.1-stable, please check if this has been fixed with this latest release.

**Comment by VRNocturne** [comment]
I updated to 4.1.1-stable and still saw the issue. 

Video clip showing the parser errors as I enter a function: https://drive.google.com/file/d/1KKl5YH3gE6FCziM_KOr7-IaJBhSi7uOI/view?usp=sharing

**Comment by rsubtil** [comment]
Thanks, I've managed to replicate the issue. I'll detail it here for someone else to tackle since GDScript stuff is not my "forte" :wink:

On the following code:
```gdscript
enum MyEnum {
	VALUE1,
	VALUE2,
}

func some_func() -> int:
	var some_var = randi_range(1, 6)
	
	match some_var:
```

- On a match value, writing `""` instead of `:` triggers a `modules/gdscript/gdscript_parser.cpp:4337 - Parser bug: Mismatch in extents tracking stack.` error:

https://github.com/godotengine/godot/assets/6501975/34e8ea14-ad71-46dd-8ba6-eb4c90810c4a

- Code completion on a return value of an enum triggers `Parser bug (please report): Trying to check compatibility of unset value type` error. If the function's return type is omitted, the error no longer occurs:

https://github.com/godotengine/godot/assets/6501975/d3f64387-d247-48b0-913d-715cfe0acb41

**Comment by ilia-abbasi** [comment]
I have the same problem (Trying to check compatibility of unset value type) on Godot v4.1.1.stable - Windows 10.0.19045.3324 - Compatibility renderer - NVIDIA GeForce 9400 GT - 8 GB RAM - Intel(R) Pentium(R) CPU G4400 3.30GHz

**Comment by JMVRy** [comment]
Same/similar issue on 4.1.2.stable.mono as well, with the following code:

```gdscript
func example() -> Vector2:
    return Vector2.ZERO
``` 

When adding a `.` to the end, or removing `.ZERO` and trying to type it back, the "Parser bug (please report)" occurs

**Comment by dalexeev** [comment]
> * On a match value, writing `""` instead of `:` triggers a `modules/gdscript/gdscript_parser.cpp:4337 - Parser bug: Mismatch in extents tracking stack.`

* Fixed by #79301.

> * Code completion on a return value of an enum triggers

> When adding a `.` to the end, or removing `.ZERO` and trying to type it back, the "Parser bug (please report)" occurs

* See #83257.
